### ISSUE TITLE
GNOME 44

### ISSUE THREAD

**Issue by cmkohnen** [SjoerdB93/Graphs#162]
GNOME 44 has been released and the accompanying platform and SDK are already available on flathub. 
I quickly did a test run and found that the current matplotlib version is incompatible with Gtk 4.10. Version 3.7 fixes that.
How did you assemble the manifests for the flatpak build? Is there a specific tool to simplify things, or is it all copied by hand?

Just opening this issue to track progress towards the new Platform, as I couldn't test any further and expect a few things have to change (colorchooser, file chooser, etc.)

**Comment by SjoerdB93** [comment]
The manifests are pretty much hand-made. There's a automated tools to create manifests which works great for most modules, but some that need specific and seperate wheels for the x86 and aarch64 build do not work (e.g. Matplotlib). Also scipy is a bit more complicated as it requires to build gfortran separately. But for future reference, that tool is flatpak-pip-generator. You can just put the modules you want in a requirement.txt, run the tool and it creates a manifest for those modules for you. There's some documentation on this [here](https://docs.flatpak.org/en/latest/python.html).

Regarding the Gnome 44 runtime, I've got things to work by changing the matplotlib lines (matplotlib itself, not the dependent modules yet) to the following, such that it installs Matplotlib 3.7.1. Indeed Matplotlib 3.6 seems to break all together on Gtk 4.10:

 ```
       {
            "type": "file",
            "url": "https://files.pythonhosted.org/packages/8a/d3/35c62c9f64ddef5f25763580a10cb1ff4a19dc1a2bf940ad06dbb10b248d/matplotlib-3.7.1-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl",
            "sha256": "56d94989191de3fcc4e002f93f7f1be5da476385dde410ddafbb70686acf00ea",
            "only-arches": "aarch64"
        },
                {
            "type": "file",
            "url": "https://files.pythonhosted.org/packages/89/f3/84a9a6613ab0d89931d785f13fa2606e03f07252875acc8ebf5b676fa3c5/matplotlib-3.7.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl",
            "sha256": "eb7d248c34a341cd4c31a06fd34d64306624c8cd8d0def7abb08792a5abfd556",
            "only-arches": "x86_64"
        }
    ]
}
```

So basically the main thing is to change the sha256 checksum as well as the url to the appropriate module version on PyPi: https://pypi.org/project/matplotlib/3.7.1/#modal-close

This runs fine (although I'll check the other modules as well), but I get an error as soon as I try to actually plot something, which I assume has to do with changes in Gtk 4.10: 

```
  File "/app/share/graphs/graphs/colorpicker.py", line 62, in update_color
    self.provider.load_from_data(css.encode())
TypeError: Gtk.CssProvider.load_from_data() takes exactly 3 arguments (2 given)
```

Will look into that. But I thought I'd reply here first so you should be able to get on as well.

**Comment by cmkohnen** [comment]
Thank you very much!
On the other hand: Is there any specific reason, we compile numpy from source instead of using a distributed wheel?

**Comment by SjoerdB93** [comment]
>  On the other hand: Is there any specific reason, we compile numpy from source instead of using a distributed wheel?

Not really. The reasoning behind this is mainly that I generated the manifests for the modules that can be generated automatically using `flatpak-pip-generator`. In the end that turned out to just be numpy, and the existing manifest (not the one in your PR) is what it came up with. The other two didn't compile nicely for different reasons, but work nicely when using the wheel.

I'm perfectly happy using the wheel for numpy as well, looking at the manifest in the PR that seems cleaner as well.
